File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.java
public class Solution {
    public boolean increasingTriplet(int[] nums) {
        var n = nums.length;
        int first = Integer.MAX_VALUE, second = Integer.MAX_VALUE;

        for(var num : nums){
            if(num <= first) first = num;           // first element of sequence found
            else if(num <= second) second = num;    // second element of sequence found
            else return true;                       // third element of sequence found
        }

        return false;
    }
}

File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        n = len(nums)
        first = second = float(inf)

        for num in nums:
            if num <= first: first = num            # first element of sequence found
            elif num <= second: second = num        # second element of sequence found
            else: return True                        # third element of sequence found
        return False